### DISCUSSION TITLE
Bad practice in cart example?

### DISCUSSION THREAD

**Discussion by hokkaido** [vuejs/pinia#1529] [Help and Questions] [answered]
I think this example propagates some questionable practices, since getters are used to derive computed values and here they mutate the state as well, as far as I can see.

https://en.wikipedia.org/wiki/Command%E2%80%93query_separation

https://github.com/vuejs/pinia/blob/08bd464ff27b7400596a1d8a78f1c43e47fc915b/packages/playground/src/stores/cart.ts#L9-L22

**Comment by posva** [comment, marked as answer]
that example doesn't mutate the state, `items` is a local variable

**Reply by hokkaido** [reply]
My bad, I need a new pair of eyes it seems.
